source/javascripts/components/unified-editor/CreateEntityDialog/CreateEntityDialog.tsx

```diff
@@ -0,0 +1,120 @@
+import { Box, Button, Dialog, DialogBody, DialogFooter, DialogProps, Input, Select } from '@bitrise/bitkit';
+import { useForm } from 'react-hook-form';
+import WorkflowService from '@/core/models/WorkflowService';
+import { useWorkflows } from '@/hooks/useWorkflows';
+import useSelectedWorkflow from '@/hooks/useSelectedWorkflow';
+import usePipelineSelector from '@/pages/PipelinesPage/hooks/usePipelineSelector';
+import PipelineService from '@/core/models/PipelineService';
+import { ENTITY_NAMES, EntityType } from '@/core/models/Entity';
+
+type FormValues = {
+  entityId: string;
+  baseEntityId: string;
+};
+
+type Props = Omit<DialogProps, 'title'> & {
+  entityType: EntityType;
+  onCreateEntity: (entityId: string, baseEntityId?: string) => void;
+};
+
+const CreateEntityDialog = ({ entityType, onClose, onCloseComplete, onCreateEntity, ...rest }: Props) => {
+  const workflows = useWorkflows();
+  const workflowIds = Object.keys(workflows);
+  const [, setSelectedWorkflow] = useSelectedWorkflow();
+
+  const { keys: pipelineIds, onSelectPipeline: setSelectedPipeline } = usePipelineSelector();
+
+  const baseIds = entityType === 'workflow' ? workflowIds : pipelineIds;
+
+  const {
+    reset,
+    register,
+    getValues,
+    handleSubmit,
+    setValue,
+    formState: { errors },
+  } = useForm<FormValues>({
+    defaultValues: {
+      entityId: '',
+      baseEntityId: '',
+    },
+  });
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const filteredValue = WorkflowService.sanitizeName(event.target.value);
+    setValue('entityId', filteredValue, {
+      shouldValidate: true,
+      shouldDirty: true,
+      shouldTouch: true,
+    });
+  };
+
+  const handleCreate = handleSubmit(({ entityId, baseEntityId }) => {
+    onCreateEntity(entityId, baseEntityId);
+    onClose();
+  });
+
+  const handleCloseComplete = () => {
+    const entityId = getValues('entityId');
+    if (entityId) {
+      if (entityType === 'workflow') {
+        setSelectedWorkflow(entityId);
+      } else {
+        setSelectedPipeline(entityId);
+      }
+    }
+    reset();
+    onCloseComplete?.();
+  };
+
+  return (
+    <Dialog
+      title={`Create ${ENTITY_NAMES[entityType]}`}
```

Comment:
I think the name of the Dialog should alse come from the specific implementation.

You should think about the generic dialog as a blueprint. Anything dynamic should be come from as a prop, and supplied by the specific implementation.

Making a generic dialog doesnt mean that a single dialog will handle all the use-cases. It will simply implement part of the dialog which would be the same for all. But the differences between the specific dialogs should be abstracted away, and the responsibility of handling those should rely on the specific implementation.

What it means the you will still have separate dialogs for Workflow and Pipeline entities, they will know their own rules, and encapsulate it inside the specific dialog. And they would pass down these implementation details to the generic dialog.

Something like this:
```js
function CreateWorkflowDialog (){
    const baseIds = ...;
    const validateName = ...;
    const onCreate = ...;

    return (
        <CreateEntity title="Create Workflow" baseIds={baseIds} validate={validateName} onCreate={onCreate} />
    );
}
```